### ISSUE TITLE
Fix false positives for idents within ::part pseudo-elements in selector-type-no-unknown

### ISSUE THREAD

**Issue by runarberg** [stylelint/stylelint#4822]
> Clearly describe the bug

There is a false positive for `selector-type-no-unknown` when selecting shadow parts of a custom element.

> Which rule, if any, is the bug related to?

`selector-type-no-unknown`

> What code is needed to reproduce the bug?

e.g.

```css
my-element::part(shadowpart) {
  /* ... */
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "selector-type-no-unknown": [
      true,
      {
        "ignore": ["custom-elements"]
      }
    ]
  }
}
```

> Which version of stylelint are you using?

e.g. `13.4.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint --ignore-path .ignore **/*.css`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```
Unexpected unknown type selector "shadowpart"   selector-type-no-unknown
```

**Comment by jeddy3** [comment]
@runarberg Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIC2BPAtDANjJMB2ALvPAA4CGATgQBQDOAFmVBAO7lUCUABMAL4gAaELRSEyADwThatQeAh4AZgEsA5lOAAdPFy6aQFAK7Za++D227d+2lhhgCECmgIoSMNHghpDeANZeLHhmXADalla6BEYwAhGRWjqRVvpqXhQwIeHghrSOSBjYuISmIAC68Va8lRVJ1Xj8vEA).

It seems the selector parser is treating the _ident_ ([see spec](https://drafts.csswg.org/css-shadow-parts/#part)) within the `::part` pseudo-element as a type selector when it shouldn't.

The rule already makes use of the [`isStandardSyntaxTypeSelector` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxTypeSelector.js):

https://github.com/stylelint/stylelint/blob/0f69eac035302bea0928e047e3395eb7164d3334/lib/rules/selector-type-no-unknown/index.js#L61-L63

So we should add a conditional to the utility to return false for nodes within `::part`.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.
